File: frontend/src/types/stock.ts
export interface Stock {
  ticker: string
  name: string
  market: string
  sector: string | null
  is_active: boolean
  latest_price: number | null
  change_pct: number | null
}

export interface StockListResponse {
  items: Stock[]
  total: number
  page: number
  per_page: number
}

export interface Fundamentals {
  market_cap: number | null
  per: number | null
  pbr: number | null
  eps: number | null
  div_yield: number | null
}

export interface StockDetailResponse extends Stock {
  fundamentals: Fundamentals | null
}

export interface DailyPriceResponse {
  date: string
  open: number
  high: number
  low: number
  close: number
  volume: number
  trading_value: number | null
  change_pct: number | null
}

File: backend/app/schemas/stock.py
from datetime import date, datetime
from pydantic import BaseModel, ConfigDict


class StockBase(BaseModel):
    ticker: str
    name: str
    market: str
    sector: str | None = None


class StockResponse(StockBase):
    is_active: bool
    latest_price: int | None = None
    change_pct: float | None = None
    trading_value: int | None = None

    model_config = ConfigDict(from_attributes=True)


class StockListResponse(BaseModel):
    items: list[StockResponse]
    total: int
    page: int
    per_page: int


class DailyPriceResponse(BaseModel):
    date: date
    open: int | None = None
    high: int | None = None
    low: int | None = None
    close: int | None = None
    volume: int | None = None
    trading_value: int | None = None
    change_pct: float | None = None

    model_config = ConfigDict(from_attributes=True)


class FundamentalsInfo(BaseModel):
    market_cap: int | None = None
    per: float | None = None
    pbr: float | None = None
    eps: int | None = None
    div_yield: float | None = None

    model_config = ConfigDict(from_attributes=True)


class StockDetailResponse(BaseModel):
    ticker: str
    name: str
    market: str
    sector: str | None = None
    is_active: bool
    latest_price: int | None = None
    change_pct: float | None = None
    fundamentals: FundamentalsInfo | None = None

    model_config = ConfigDict(from_attributes=True)
